server/src/main/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapper.java

```diff
@@ -452,7 +461,12 @@ public LegacyGeoShapeFieldMapper(String simpleName, MappedFieldType mappedFieldT
         super(simpleName, mappedFieldType, Collections.singletonMap(mappedFieldType.name(), Lucene.KEYWORD_ANALYZER),
             builder.ignoreMalformed.get(), builder.coerce.get(), builder.ignoreZValue.get(), builder.orientation.get(),
             multiFields, copyTo, indexer, parser);
+        if (builder.indexCreatedVersion.onOrAfter(Version.V_8_0_0)) {
```

Comment:
Can we move this into the Builder constructor? The earlier we throw the better, in general.

Reply:
I have tried but I have issues with this test:

https://github.com/elastic/elasticsearch/blob/91c700bd99e59d099cee282fe5b4a1a266f70748/test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java#L270

It seems that checking of unknown parameters happen after the builder.